bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -614,8 +614,24 @@ public void safeRun() {
         }
     }
 
-    void checkForFaultyBookies() {
+    void checkForFaultyBookies()  {
         List<BookieId> faultyBookies = bookieClient.getFaultyBookies();
+        if (faultyBookies.isEmpty()) {
+            return;
+        }
+
+        boolean isEnable = true;
+        try {
+            isEnable = metadataDriver.isHealthCheckEnabled().get();
+        } catch (Exception e) {
```

Comment:
Please don't catch raw Exception.
If reading from ZK fails it is likely that we won't be able to move forward in the following lines, so it is better to easly exist.

Reply:
OK, I will fixed